enigma-principal/app/src/boot_network/principal_manager.rs

```diff
@@ -64,6 +70,52 @@ pub struct PrincipalManager {
     pub eid: sgx_enclave_id_t,
 }
 
+pub struct SgxEthereumSigner {
+    eid: sgx_enclave_id_t,
+}
+
+impl SgxEthereumSigner {
+    pub fn new(eid: sgx_enclave_id_t) -> SgxEthereumSigner {
+        SgxEthereumSigner{ eid }
+    }
+}
+
+impl EcdsaSign for SgxEthereumSigner {
+    fn sign_hashed(&self, to_sign: &[u8; 32]) -> [u8; 65] {
```

Comment:
As explained, this PR makes it so that whenever a key is not provided as part of the configuration, a key is instead generated in the enclave.
We should document this, but this isn't the right location. perhaps in the configuration struct used by KM? @Cashmaney ?